.github/workflows/publish-docs.yml

```diff
@@ -0,0 +1,71 @@
+name: Publish docs to GitHub pages
+
+on:
+  push:
+    branches:
+      - 'docs/**'
+    paths:
+      - 'docs/**'
+
+jobs:
+  publish:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: [3.7]
+        package: [docs]
+
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          # Only a single commit is fetched by default, for the ref/SHA that triggered the workflow.
+          # Set fetch-depth: 0 to fetch all history for all branches and tags.
+          fetch-depth: 0
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Set the current branch environment variable
+        working-directory: ${{ matrix.package }}
+        run: |
+          CURRENT_BRANCH_VAR=${GITHUB_REF#refs/heads/}
+          echo "Current branch: $CURRENT_BRANCH_VAR"
+          # Set CURRENT_BRANCH environment variable to the current branch name.
+          echo "::set-env name=CURRENT_BRANCH::$(echo $CURRENT_BRANCH_VAR)"
+      - name: Display all remote branches
+        working-directory: ${{ matrix.package }}
+        run: |
+          # Display all remote branches
+          git branch -r
+      - name: Set the latest docs branch environment variable
+        working-directory: ${{ matrix.package }}
+        run: |
+          # Get only docs branches, extract the "docs/x.y.z" part, sort them in descending order, and get the first one.
+          LATEST_DOCS_BRANCH_VAR=$(git branch -r | grep -e ".*\/*docs\/[0-9].[0-9].[0-9]" | sed -n "s/.*\/*\(docs\/[0-9].[0-9].[0-9]\).*/\1/p" | sort -r | head -n 1)
+          echo "Latest docs branch: $LATEST_DOCS_BRANCH_VAR"
+          # Set the LATEST_DOCS_BRANCH environment variable.
+          echo "::set-env name=LATEST_DOCS_BRANCH::$(echo $LATEST_DOCS_BRANCH_VAR)"
+      - name: Check that the current branch is the latest docs branch, fail otherwise
+        working-directory: ${{ matrix.package }}
+        run: |
+          if [ $CURRENT_BRANCH != $LATEST_DOCS_BRANCH ]; then
+            exit 1
+          else
+            echo "The action is running on the latest docs branch."
+          fi
+      - name: Install Pipenv
+        uses: dschep/install-pipenv-action@v1
+      - name: Install dependencies for building documenation
+        working-directory: ${{ matrix.package }}
+        run: |
+          pipenv install
+      - name: Build documentation
+        working-directory: ${{ matrix.package }}
+        run: |
+          pipenv run mkdocs build --clean
+      - name: Deploy the contents of docs/site to gh-pages 🚀
+        uses: JamesIves/github-pages-deploy-action@releases/v3
```

Comment:
this looks like a branch that keeps getting updated as new 3.x releases go out, should we pin it to a tag instead?

Reply:
We can pin it to the major version of a release: `uses: JamesIves/github-pages-deploy-action@v3`.

Reply:
Actually, they might not publish releases, . I tried using `uses: JamesIves/github-pages-deploy-action@v3` and I ran into the following error:

`An action could not be found at the URI 'https://api.github.com/repos/JamesIves/github-pages-deploy-action/tarball/v3'`

If we pin the action to a tag, we won't automatically get compatible upgrades. I think we might have to use the release branch or use a different action, maybe this one https://github.com/marketplace/actions/github-pages-action.

Reply:
was looking at [https://github.com/marketplace/actions/github-pages-action](https://github.com/marketplace/actions/github-pages-action)
is that more popular / stable? which one do you think we should use?

also looking through some of the examples in that, it always does:
```
- name: Deploy
        uses: peaceiris/actions-gh-pages@v3
```
so looks like even they recommend using only the major version, which resolves this comment.
I do think the one on the marketplace might be worth considering, if you think it has equivalent or better functionality.

Reply:
I switched to using https://github.com/marketplace/actions/github-pages-action as suggested.